### PR TITLE
Remove `panics` on the inv methods

### PR DESCRIPTION
**Pull request by josepchetrit12**
I removed the `panics` in the inv methods and returned the `DivisionByZero` error, this PR is related to [#77](https://github.com/lambdaclass/lambdaworks/issues/77) issue.

### CODE REVIEW

**Review by MauroToscano — CHANGES_REQUESTED** [review]
Inv Error Handling looks good. But it shouldn't be propagated through Ops. Div op can panic, and the developer assumes that risk. The operator should say it can panic and when in a 'Panics' section in the documentation. 

We need to create checked_ops, so the users have a safe way to use the operators, but we will create another issue for that.

**Comment by josepchetrit12** [comment]
OK @MauroToscano I will change this!

**Comment by josepchetrit12** [comment]
@MauroToscano, Now the div operation can panic!

File: math/src/field/fields/u384_prime_field.rs
Comment (Oppen) [inline math/src/field/fields/u384_prime_field.rs:83]:
```suggestion
        let inv_b = Self::inv(b).expect("Division by zero!");
        Self::mul(a, &inv_b)
```

File: math/src/field/fields/u64_prime_field.rs
Comment (Oppen) [inline math/src/field/fields/u64_prime_field.rs:35]: About the same as previous suggestion.

**Review by Oppen — APPROVED** [review]
I left a few style comments, but none of them critical.

File: math/src/field/fields/u384_prime_field.rs
Comment (ilitteri) [inline math/src/field/fields/u384_prime_field.rs:112]: Could we remove this extra line?

**Review by ilitteri — APPROVED** [review]
LGTM, we could merge after solving the conflicts

**Review by MauroToscano — CHANGES_REQUESTED** [review]

**Comment by MauroToscano** [comment]
> @MauroToscano, Now the div operation can panic!

Looking good! I have left one more comment, and it's good to be merged for me.

File: math/src/fft/fft_cooley_tukey.rs
Comment (MauroToscano) [inline math/src/fft/fft_cooley_tukey.rs:56]: Is this as_ref needed ? The multiplication of field elements is defined for all combinations of & and values.

File: math/src/fft/fft_cooley_tukey.rs
Comment (josepchetrit12) [reply math/src/fft/fft_cooley_tukey.rs:53]: > This function should propagate the error, since the user of the function could send a 0

One question this function should propagate the error or can panic?
```
pub fn inverse_fft<F: IsField + IsTwoAdicField>(
    evaluations: &[FieldElement<F>],
) -> Result<Vec<FieldElement<F>>, FFTError> {
    let omega = F::get_root_of_unity(log2(evaluations.len())?)?;
    Ok(inverse_cooley_tukey(evaluations, omega))
}
```

Comment (MauroToscano) [reply math/src/fft/fft_cooley_tukey.rs:53]: Yes. The criteria is no function can panic, except the std ops, in particular the div. This is the typical behavior of the commonly used div operator "/". We still need checked ops, so people can have a non panicking alternative, but that's another issue.

Here you can add an FFTError, and use both in the inverse and here

Comment (josepchetrit12) [reply math/src/fft/fft_cooley_tukey.rs]: @MauroToscano Can you review my last commit? Now the function can propagate the error, but I would like to know if the option I did is correct. Because in `inverse_fft` function can return two types of errors `FieldError` or `FFTError`.

**Review by jrchatruc — APPROVED** [review]

File: math/src/fft/fft_cooley_tukey.rs
Comment (MauroToscano) [reply math/src/fft/fft_cooley_tukey.rs]: It's returning an FFTError, so I'm not sure what do you mean. A function can't return two types of errors unless you are boxing them, and I'm not seeing that.

Comment (josepchetrit12) [reply math/src/fft/fft_cooley_tukey.rs]: I know it is returning a `FFTError`, but inside this error I added the `FieldError` to it, you can check it in this [file](https://github.com/lambdaclass/lambdaworks/blob/642d80f7c89b003ee5ef7a606e113cbc87d2feb5/math/src/fft/errors.rs#L12) 

I did this because `F::get_root_of_unity(log2(evaluations.len())?)?` can return a `FFTError` and `inverse_cooley_tukey(evaluations, omega)` can return a `FieldError`.

Comment (Oppen) [reply math/src/fft/fft_cooley_tukey.rs]: @MauroToscano a quick review? This has 3 other approvals and is blocked by this comment only.

**Comment by MauroToscano** [comment]
This is a good PR, but it's too outdated. I'm closing it, we can merge it if it's updated

**Pull request closed without merging**
